ironmon_tracker/Battle.lua

```diff
@@ -507,4 +602,118 @@ function Battle.changeOpposingPokemonView(isLeft)
+				Tracker.TrackAbility(abilityOwner.pokemonID, Battle.BattleAbilities[targetTeamIndex][targetSlot].ability)
+			end
+
+			Battle.BattleAbilities[attackerTeamIndex][attackerSlot].abilityOwner.isOwn = Battle.BattleAbilities[targetTeamIndex][targetSlot].abilityOwner.isOwn
+			Battle.BattleAbilities[attackerTeamIndex][attackerSlot].abilityOwner.slot = Battle.BattleAbilities[targetTeamIndex][targetSlot].abilityOwner.slot
+			Battle.BattleAbilities[attackerTeamIndex][attackerSlot].ability = Battle.BattleAbilities[targetTeamIndex][targetSlot].ability
+
+			--Track Transform changes
+			if moveUsed == 144 then
+				Battle.BattleAbilities[attackerTeamIndex][attackerSlot].transformData.isOwn = Battle.BattleAbilities[targetTeamIndex][targetSlot].transformData.isOwn
+				Battle.BattleAbilities[attackerTeamIndex][attackerSlot].transformData.slot = Battle.BattleAbilities[targetTeamIndex][targetSlot].transformData.slot
+			end
+
+		end
+	elseif ability ~= nil and ability ~=0 then
+		if ability == 36 then --Trace
+			-- In double battles, Trace picks a random target, so we need to grab the battle index from the text variable, gBattleTextBuff1[2]
+			local tracerTeamIndex = Battle.battler % 2
+			local tracerTeamSlot = Battle.Combatants[Battle.IndexMap[Battle.battler]]
+			local target = Memory.readbyte(GameSettings.gBattleTextBuff1 + 2)
+			local targetTeamIndex = target % 2
+			local targetTeamSlot = Battle.Combatants[Battle.IndexMap[target]]
+
+			--Track Trace here, otherwise when we try to track normally, the pokemon's battle ability and owner will have already updated to what was traced.
+			local abilityOwner = Tracker.getPokemon(Battle.BattleAbilities[tracerTeamIndex][tracerTeamSlot].abilityOwner.slot,Battle.BattleAbilities[tracerTeamIndex][tracerTeamSlot].abilityOwner.isOwn)
+			Tracker.TrackAbility(abilityOwner.pokemonID, ability)
+			
+			Battle.BattleAbilities[tracerTeamIndex][tracerTeamSlot].abilityOwner.isOwn = Battle.BattleAbilities[targetTeamIndex][targetTeamSlot].abilityOwner.isOwn
+			Battle.BattleAbilities[tracerTeamIndex][tracerTeamSlot].abilityOwner.slot = Battle.BattleAbilities[targetTeamIndex][targetTeamSlot].abilityOwner.slot
+			Battle.BattleAbilities[tracerTeamIndex][tracerTeamSlot].ability = Battle.BattleAbilities[targetTeamIndex][targetTeamSlot].ability
+		end
+	end
+end
+
+function Battle.resetAbilityMapPokemon(slot, isOwn)
+	local teamIndex = Utils.inlineIf(isOwn,0,1)
+	Battle.BattleAbilities[teamIndex][slot].abilityOwner.isOwn = isOwn
+	Battle.BattleAbilities[teamIndex][slot].abilityOwner.slot = slot
+	Battle.BattleAbilities[teamIndex][slot].ability = Battle.BattleAbilities[teamIndex][slot].originalAbility
+
+	Battle.BattleAbilities[teamIndex][slot].transformData.isOwn = isOwn
+	Battle.BattleAbilities[teamIndex][slot].transformData.slot = slot
+end
+
+function Battle.handleTransformedMons()
```

Comment:
Nit-picky request here, but could this function be renamed to describe what it's doing? I'm not entirely sure what "handling a transformed mon" means under the context of a battle. Is the intent of this function: `trackTransformedMoves()` ?

Reply:
Can update the name and yes, this function is two-part
1) it determines if an allied pokemon is selecting a move to use
2) if it is, and we are transformed into an Enemy pokemon, we track all of that enemy's moves